### PR TITLE
MM-1770 Add the ability to cancel in-progress file uploads

### PR DESCRIPTION
**Pull request by hmhealey**
To see a diff without the cosmetic refactoring, go to https://github.com/hmhealey/mattermost-platform/compare/20d755f%5E...96ad8cb

### CODE REVIEW

**Comment by jwilander** [comment]
+1

File: web/react/components/create_comment.jsx [hunk 1]
```diff
@@ -40,163 +44,194 @@ module.exports = React.createClass({
...
+        draft['uploadsInProgress'] = draft['uploadsInProgress'].concat(clientIds);
+        PostStore.storeCommentDraft(this.props.rootId, draft);
 
-        draft.previews = previews;
-        draft.uploadsInProgress = num-1;
+        this.setState({uploadsInProgress: draft['uploadsInProgress']});
+    },
+    handleFileUploadComplete: function(filenames, clientIds, channelId) {
+        var draft = PostStore.getCommentDraft(this.props.rootId);
+        if (!draft) {
```
Comment (crspeller) [inline web/react/components/create_comment.jsx]: Should PostStore.getDraft not do this itself? I see this copy code quite a bit.

Comment (jwilander) [inline web/react/components/create_comment.jsx]: Agreed, this code is repeated a lot

Comment (hmhealey) [inline web/react/components/create_comment.jsx]: I was definitely thinking of that when I was writing this. I need to make sure that there's nowhere that depends on the behaviour of returning null, but I'll add that in.

**Comment by hmhealey** [comment]
Added in a default, empty draft instead of just returning null from the PostStore when no draft exists. Thanks Chris for the reminder about that.

**Pull request merged**